packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -90,15 +90,39 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
             this.driverPolicies.enableRestLess,
             resolvedUrl.endpoints.ordererUrl,
         );
+
         // the backend responds with the actual document ID associated with the new container.
-        const documentId = await ordererRestWrapper.post<string>(
+        const res = await ordererRestWrapper.post<{ id: string, token?: string } | string>(
             `/documents/${tenantId}`,
             {
                 summary: convertSummaryToCreateNewSummary(appSummary),
                 sequenceNumber: documentAttributes.sequenceNumber,
                 values: quorumValues,
+                generateToken: true,
             },
         );
+
+        // For supporting backward compatibility, when the request has generateToken === true, it will return
+        // an object instead of string
+
+        let documentId: string;
+        let token: string | undefined;
+
+        if (typeof res === "string") {
+            documentId = res;
+        } else {
+            documentId = res.id;
+            token = res.token;
+        }
+
+        if (token && this.tokenProvider.documentPostCreateCallback !== undefined) {
+            try {
+                await this.tokenProvider.documentPostCreateCallback(documentId, token);
+            } catch (error) {
+                logger2.sendErrorEvent({ eventName: "catchDocumentPostCreateCallback" }, error);
```

Comment:
By catching the error here, if the `documentPostCreateCallback` throws an error (which I'm guessing would be the behavior if the token is invalid for the current user), then continuing will not be blocked. You could/should probably just remove this try..catch and event log if the intent is to fail here and prevent user from continuing.

Reply:
I would expect the callback failure to be handled by the user's implementation (e.g. adding a try..catch within their own implementation, that parses the error and rethrows if it is blocking)